File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    Set<Integer> set = new HashSet<>();
    char[] arr;
    boolean[] isVisited;
    public int solution(String numbers) {
        int answer = 0;
        arr = numbers.toCharArray();
        isVisited = new boolean[arr.length];

        dfs("", 0);

        return set.size();
    }

    public void dfs(String str, int depth) {
        if(str != "") {
            int num = Integer.parseInt(str);
            if(isPrime(num)) {
                set.add(num);
            }
        }
        if(depth >= arr.length) return;
        for (int i = 0; i < arr.length; i++) {
            if(isVisited[i]) continue;
            isVisited[i] = true;
            dfs(str + arr[i], depth + 1);
            isVisited[i] = false;
        }
    }

    public boolean isPrime(int num) {
        if(num == 1 || num == 0) return false;
        for(int x = 2; x <= Math.sqrt(num); x++) {
            if(num % x == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def solution(numbers):
    answer = []
    numbers = [n for n in numbers]
    per = []
    for i in range(1, len(numbers)+1):
        per += list(permutations(numbers, i))
    new_nums = [int("".join(p)) for p in per]
    
    for i in new_nums:
        if i < 2:
            continue
        is_prime = True
        for n in range(2, int(i**0.5)+1):
            if i % n == 0:
                is_prime = False
                break
        if is_prime:
            answer.append(i)
    
    return len(set(answer))